foursight_core/app_utils.py

```diff
@@ -467,7 +503,111 @@ def view_foursight(self, environ, is_admin=False, domain="", context="/"):
         html_resp.status_code = 200
         return self.process_response(html_resp)
 
-    def view_foursight_check(self, environ, check, uuid, is_admin=False, domain="", context="/"):
+    # dmichaels/2020-08-01:
+    # Added /view/info for debugging/troubleshooting purposes.
+    def view_info(self, request, is_admin=False, domain="", context="/"):
+
+        def sorted_dict(dictionary: dict) -> dict:
+            result = {}
+            for key in sorted(dict(dictionary).keys(), key=lambda key: key.lower()):
+                result[key] = dictionary[key]
+            return result
+
+        def get_obfuscated_aws_credentials_info() -> dict:
+            try:
+                session = boto3.session.Session()
+                credentials = session.get_credentials()
+                access_key_id = credentials.access_key
+                secret_access_key = credentials.secret_key
+                region_name = session.region_name
+                session_token = credentials.token
+                caller_identity = boto3.client("sts").get_caller_identity()
+                user_arn = caller_identity["Arn"]
+                account_number = caller_identity["Account"]
+                aws_credentials_info = {
+                    "AWS Account Number:": account_number,
+                    "AWS User ARN:": user_arn,
+                    "AWS Access Key ID:": access_key_id,
+                    "AWS Secret Access Key:": obfuscate(secret_access_key),
+                    "AWS Region Name:": region_name,
+                    "AWS Session Token:": obfuscate(session_token)
```

Comment:
I think including anything about the secret key or session token is questionable, probably not necessary.